crates/runtime/src/datafusion/query.rs

```diff
@@ -290,20 +310,24 @@ fn attach_query_context_to_stream(
     let schema_copy = Arc::clone(&schema);
 
     let mut num_records = 0u64;
-    let mut ctx = ctx;
 
     let updated_stream = stream! {
         while let Some(batch_result) = stream.next().await {
 
             match &batch_result {
                 Ok(batch) => {
                     num_records += batch.num_rows() as u64;
+                    yield batch_result
                 }
                 Err(e) => {
-                    ctx = ctx.error(e.to_string());
+                    ctx
+                    .schema(schema_copy)
+                    .rows_produced(num_records)
+                    .finish_with_error(e.to_string(), ErrorCode::DataReadError).await;
+                    yield batch_result;
+                    return;
```

Comment:
is this return intentional?

Reply:
does it change existing behaviour?

Reply:
Confirm this is intentional.

Returning Err as record batch finalizes the stream and after while section is not executed. This is used to tell rust that we are not going to use `ctx` anymore and can safely consume it (`ctx` is also used below to finish query for happy path).